plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMaterializedViewDefinition.java

```diff
@@ -0,0 +1,187 @@
+        this.owner = requireNonNull(owner, "owner is null");
+
+        if (catalog.isEmpty() && schema.isPresent()) {
+            throw new IllegalArgumentException("catalog must be present if schema is present");
+        }
+        if (columns.isEmpty()) {
+            throw new IllegalArgumentException("columns list is empty");
+        }
+    }
+
+    public static IcebergMaterializedViewDefinition fromConnectorMaterializedViewDefinition(ConnectorMaterializedViewDefinition definition)
+    {
+        return new IcebergMaterializedViewDefinition(
+                definition.getOriginalSql(),
+                definition.getCatalog(),
+                definition.getSchema(),
+                definition.getColumns().stream()
+                        .map(column -> new Column(column.getName(), column.getType()))
+                        .collect(toImmutableList()),
+                definition.getComment(),
+                definition.getOwner());
+    }
+
+    @JsonProperty
+    public String getOriginalSql()
+    {
+        return originalSql;
+    }
+
+    @JsonProperty
+    public Optional<String> getCatalog()
+    {
+        return catalog;
+    }
+
+    @JsonProperty
+    public Optional<String> getSchema()
+    {
+        return schema;
+    }
+
+    @JsonProperty
+    public List<Column> getColumns()
+    {
+        return columns;
+    }
+
+    @JsonProperty
+    public Optional<String> getComment()
+    {
+        return comment;
+    }
+
+    @JsonProperty
+    public String getOwner()
+    {
+        return owner;
+    }
+
+    @Override
+    public String toString()
+    {
+        StringJoiner joiner = new StringJoiner(", ", "[", "]");
+        joiner.add("originalSql=[" + originalSql + "]");
+        catalog.ifPresent(value -> joiner.add("catalog=" + value));
+        schema.ifPresent(value -> joiner.add("schema=" + value));
+        joiner.add("columns=" + columns);
+        comment.ifPresent(value -> joiner.add("comment=" + value));
+        joiner.add("owner=" + owner);
+        return getClass().getSimpleName() + joiner;
+    }
+
+    public static final class Column
+    {
+        private final String name;
+        private final TypeId type;
+
+        @JsonCreator
+        public Column(
+                @JsonProperty("name") String name,
+                @JsonProperty("type") TypeId type)
+        {
+            this.name = requireNonNull(name, "name is null");
+            this.type = requireNonNull(type, "type is null");
+        }
+
+        @JsonProperty
+        public String getName()
+        {
+            return name;
+        }
+
+        @JsonProperty
+        public TypeId getType()
+        {
+            return type;
+        }
+
+        @Override
+        public String toString()
+        {
+            return name + " " + type;
+        }
+    }
+
+    public static String encodeMaterializedViewData(IcebergMaterializedViewDefinition definition)
```

Comment:
move this methods to the top of class along with `fromConnectorMaterializedViewDefinition` method